trio/_sync.py

```diff
@@ -823,21 +823,23 @@ class Queue:
       capacity (int): The maximum number of items allowed in the queue before
           :meth:`put` blocks. Choosing a sensible value here is important to
           ensure that backpressure is communicated promptly and avoid
-          unnecessary latency. If in doubt, use 1.
+          unnecessary latency. If in doubt, use 0.
 
     """
 
     def __init__(self, capacity):
         if not isinstance(capacity, int):
             raise TypeError("capacity must be an integer")
-        if capacity < 1:
-            raise ValueError("capacity must be >= 1")
-        # Invariants:
-        #   get_semaphore.value() == len(self._data)
-        #   put_semaphore.value() + get_semaphore.value() = capacity
+        if capacity < 0:
+            raise ValueError("capacity must be >= 0")
         self.capacity = operator.index(capacity)
-        self._put_semaphore = Semaphore(capacity, max_value=capacity)
-        self._get_semaphore = Semaphore(0, max_value=capacity)
+        # {task: abort func}
+        self._get_wait = OrderedDict()
```

Comment:
I don't understand why this has the abort fn in it. It looks like all the abort fn does is remove the task from `self._get_wait`, and the only place we use it is in `put`, which calls it... immediately after removing the task from `self._get_wait`, so it's a no-op. And then we have a `try`/`except` in the `get` abort fn just to work around this?

Can we make `_get_wait` a mapping `{task: None}`, i.e. basically just an ordered set? Or is there a subtlety I'm missing?

Reply:
On further thought: I bet this is left-over from my `multi_get` hack, huh. I think it's unnecessary here now though...